src/diffusers/models/modeling_utils.py

```diff
@@ -315,7 +316,9 @@ def save_pretrained(
         weights_name = _add_variant(weights_name, variant)
         weight_name_split = weights_name.split(".")
         if len(weight_name_split) in [2, 3]:
```

Comment:
```suggestion
```
we don't need weight_name_split now, no?

Reply:
We don't need it but it's a sanity check suggested by @Wauplin when we integrated `split_torch_state_dict_into_shards()`. Okay without it but I don't think it's a problem to have it either.

Reply:
This check is indeed optional. I suggested to add it because I was not 100% sure how the weights naming was working. You definitely need the `weights_name_pattern` though